eqt/ui/FormDialog.py

```diff
@@ -115,30 +108,86 @@ def addSpanningWidget(self, qwidget, name=None, layout='form'):
             raise ValueError(
                 f"layout {layout} is not recognised, must be set to 'form' or 'vertical'")
 
+    def insertWidgetToFormLayout(self, row, name, qwidget, qlabel=None):
+        '''
+        Inserts a widget and a label widget, or a spanning widget if 'qlabel' is None, to the form layout
+        in the position specified by row. If row is out of bounds, the widget is added at the end.
+        It adds to the widget dictionary, the widget number dictionary, and the default widget states dictionary.
+
+        Parameters:
+        ----------
+        row: int
+        name: str
+        qwidget: qwidget
+        qlabel: qlabel widget or str
+        '''
+        self.formWidget.insertWidgetToFormLayout(row, name, qwidget, qlabel)
+
+    def insertWidgetToVerticalLayout(self, row, qwidget):
+        '''
+        Inserts a widget to the vertical layout at position specified by row.
+        '''
+        self.formWidget.uiElements['verticalLayout'].insertWidget(row, qwidget)
+
+    def removeWidgetFromVerticalLayout(self, qwidget):
+        '''
+        Removes a widget from the vertical layout.
+        '''
+        self.formWidget.uiElements['verticalLayout'].removeWidget(qwidget)
+        qwidget.setParent(None)
+
+    def getWidgetFromVerticalLayout(self, index):
+        '''
+        Returns the widget in the vertical layout located at position index.'''
+        return self.formWidget.uiElements['verticalLayout'].itemAt(index).widget()
+
+    def removeWidget(self, name):
+        '''
```

Comment:
```suggestion
        '''Removes the widget with name `name` from the widgets in the `FormDialog`
        
```

Reply:
Please can you add a sentence like this at the beginning?

What this method does is clearly much more complex and it is well described later.

Reply:
Yes, thanks for spotting this.